README.md
# [QCBS R Workshop series](https://wiki.qcbs.ca/r)

This series of 10 workshops walks participants through the steps required to use R for a wide array of statistical analyses relevant to research in biology and ecology. These open-access workshops were created by members of the QCBS both for members of the QCBS and the larger community.

The content of this workshop has been peer-reviewed by several QCBS members. If you would like to suggest modifications, please contact the current series coordinators, listed [here](https://wiki.qcbs.ca/r) or open a pull request (see contributing guidelines at https://qcbsrworkshops.github.io/guidelines.html).

# [Série d'ateliers R du CSBQ](https://wiki.qcbs.ca/r)

Cette série de 10 ateliers guide les participants à travers les étapes requises afin de maîtriser le logiciel R pour une grande variété d’analyses statistiques pertinentes en recherche en biologie et en écologie. Ces ateliers en libre accès ont été créés par des membres du CSBQ à la fois pour les membres du CSBQ et pour la grande communauté d’utilisateurs de R.

Le contenu de cet atelier a été révisé par plusieurs membres du CSBQ. Si vous souhaitez y apporter des modifications, veuillez SVP contacter les coordonnateurs actuels de la série, listés [ici](https://wiki.qcbs.ca/r)
ou ouvrez un pull request (voir les instructions https://qcbsrworkshops.github.io/guidelines.html).



# Workshop 1: Introduction to R
[![Build Status](https://img.shields.io/travis/QCBSRworkshops/workshop01/dev?style=flat-square&logo=travis)](https://travis-ci.org/QCBSRworkshops/workshop01)
[![badge](https://img.shields.io/static/v1?style=flat-square&label=repo&message=dev&color=6f42c1&logo=github)](https://github.com/QCBSRworkshops/workshop01)
[![badge](https://img.shields.io/static/v1?style=flat-square&label=wiki&message=01&logo=wikipedia)](https://wiki.qcbs.ca/r_workshop1)
[![badge](https://img.shields.io/static/v1?style=flat-square&label=Slides&message=01&color=red&logo=html5)](https://qcbsrworkshops.github.io/workshop01/workshop01-en/workshop01-en.html)
[![badge](https://img.shields.io/static/v1?style=flat-square&label=Slides&message=01&color=red&logo=adobe-acrobat-reader)](https://qcbsrworkshops.github.io/workshop01/workshop01-en/workshop01-en.pdf)
[![badge](https://img.shields.io/static/v1?style=flat-square&label=script&message=01&color=2a50b8&logo=r)](https://qcbsrworkshops.github.io/workshop01/workshop01-en/workshop01-en.R)

In this introductory R Workshop you will learn what R open-source statistical software is, why you should absolutely start using it, and all the first steps to help you get started in R. We will show you how R can act as a calculator, teach you about the various types of objects in R, show you how to use functions and load packages, and find all the resources you need to get help. If any of this sounds obscure, don’t worry! By the end of this workshop you’ll know what all these words mean!

# Atelier 1: Introduction à R
[![Build Status](https://img.shields.io/travis/QCBSRworkshops/workshop01/dev?style=flat-square&logo=travis)](https://travis-ci.org/QCBSRworkshops/workshop01)
[![badge](https://img.shields.io/static/v1?style=flat-square&label=repo&message=dev&color=6f42c1&logo=github)](https://github.com/QCBSRworkshops/workshop01)
[![badge](https://img.shields.io/static/v1?style=flat-square&label=wiki&message=01&logo=wikipedia)](https://wiki.qcbs.ca/r_atelier1)
[![badge](https://img.shields.io/static/v1?style=flat-square&label=Diapos&message=01&color=red&logo=html5)](https://qcbsrworkshops.github.io/workshop01/workshop01-fr/workshop01-fr.html)
[![badge](https://img.shields.io/static/v1?style=flat-square&label=Diapos&message=01&color=red&logo=adobe-acrobat-reader)](https://qcbsrworkshops.github.io/workshop01/workshop01-fr/workshop01-fr.pdf)
[![badge](https://img.shields.io/static/v1?style=flat-square&label=script&message=01&color=2a50b8&logo=r)](https://qcbsrworkshops.github.io/workshop01/workshop01-fr/workshop01-fr.R)

Dans cet atelier d’introduction à R, vous apprendrez ce qu’est le logiciel d’analyse statistique R, vous verrez plusieurs raisons pourquoi vous devriez absolument l’utiliser (!), et vous ferez vos premiers pas dans ce logiciel. Vous verrez comment R peut être utilisé comme une calculatrice, vous apprendrez ce qu’est un « objet » dans R, et vous utiliserez des fonctions simples. De plus, vous apprendrez comment télécharger de nouvelles fonctions et comment trouver des ressources d’aide pour utiliser R. Si tout ceci vous semble étrange, ne vous inquiétez pas! À la fin de cet atelier, vous comprendrez tous ces concepts.



NEWS.md
# development version

* Update README. 

# version 0.1.0

* Add first slides.
* Fix a bunch of typos.
* Remove iframes.
* Update template.
* Remove `pkgs.yaml`.

# version 0.0.1

* Use Rmarkdown over Prezi as presentation support.

workshop01-fr/ReferenceScriptWorkshop1.R
#### QCBS Workshop 1 
#### R reference script 
## September 2018 

## Challenge 1
## How to read the console
output <- "This is the output"
#output <- "Ceci est la sortie"
output

# Using R as a calculator
## Addition, substraction
1 + 1
10 - 1 

## Multiplications and Divisions
2 * 2
8 / 2

## Exponents 
2^3

## Challenge 2 
# Use R to calculate the following testing question.

2 + 16 * 24 - 56 

## Challenge 3
# Use R to calculate the following testing question.
2 + 16 * 24 - 56 / (2 + 1) - 457

## Challenge 4

# What is the area of a circle, with a radius of 5cm?
3.1416 * 5^2
# Note that R has some built-in constants such as pi,therefore: 
pi * 5^2

## Objects 

# Objects are one of the most useful concepts in R. 
# You can store values as named objects using the assigment operator "<-"

object_name <- "assigned Value" 
# nom_objet <- "valeurs"

## Objects naming: good practices 

# Try to have short and explicit names 
# Adding spaces before the "<-" is recommended 
# When typing the object names, R will return its value 

mean.x <- (2 + 6) / 2
mean.x 

## Challenge 5 
# Create an object with a value of 1+1.718282 (Euler's  number) and name it "euler.value"

euler_value <- 1 + 1.718282
euler_value

## Challenge 6 
# Create a second object with a name that starts with a number, What happens?

## Types of data structures in R 

# - Vectors
# - Data frames
# - Matrices, arrays and lists

## Vectors

# - An entity consisting of a list of related values
# - A single value is called an *atomic value*
#   - All values of a vector must have the **same mode** (or class).
# * Numeric: only numbers
# * Logical: True/False entries
# * Character: Text, or a mix of text and other modes

# Creating vectors require the c() function // c() stands for combine or concatenate 

vector <- c("value1", "value2")

## Numeric vectors 
num.vector <- c(1,4,3,98,32,-76, -4)
num.vector 

## Character vectors 
char.vector <- c("blue", "red", "green")
char.vector

## Logical vectors 
bool.vector <- c(TRUE, TRUE, FALSE)
bool.vector

bool.vector2 <- c(T,T,F)
bool.vector2

## Challenge 7 
# Create a vector containing the first 5 odd numbers, starting from 1, and name it "odd.n"
odd.n <- c(1,3,5,7)

## We can use vectors for calculations 
x <- c(1:5)
y <- 6 

x+y
x*y 

## Data frames 

# - Used to store data tables
# - A list of vectors of the same length
# - Columns = variables
# - Rows = observations, sites, cases, replicates, ...
# - Differents columns can have different modes

## One way to create vectors 
# Start by creating vectors 

siteID <- c("A1.01", "A1.02", "B1.01", "B1.02")
soil_pH <- c(5.6, 7.3, 4.1, 6.0)
num.sp <- c(17, 23, 15, 7)
treatment <- c("Fert", "Fert", "No_fert", "No_fert")

# We then combine them using the function data.frame() 
my.first.df <- data.frame(siteID, soil_pH, num.sp, treatment)
my.first.df

## Matrices, Arrays and Lists 

## Indexing vectors 
# You  can use indexing to chose a particular position, 
# let's say we want to see the second value of our `odd.n` vector

odd.n[2]
# It also work with multiple positions: 
odd.n[c(2,4)]
# It can be used to remove some values at particular positions 
odd.n[-c(1,2)]

# If you select a position that is not in the vector: 
odd.n[c(1,5)]

# You can also use conditions to select values 
char.vector[char.vector == "blue"]

## Challenge 8 

# Using the vector "num.vector"
# - Extract the 4th value
# - Extract the 1st and 3rd values
# - Extract all values except for the 2nd and the 4th

num.vector[4]
num.vector[c(1,3)]
num.vector[c(-2,-4)]

## Challenge 9 
# Explore the difference between these 2 lines of code:
char.vector == "blue"
char.vector[char.vector == "blue"]

## Indexing data frames
## Challenge 10 

# 1. Extract the `num.sp` column from `my.first.df` and multiply its value by  the first four values of `num.vec`.

my.first.df$num.sp * num.vector[c(1:4)]
# or
my.first.df[,3] * num.vector[c(1:4)]

# 2. After that, write a statement that checks if the values you obtained are greater than 25.
(my.first.df$num.sp * num.vector[c(1:4)]) > 25

## Functions 

# A function is a tool to simplify your life.
# 
# It allows you to quickly execute operations on objects without having to write every mathematical step.
# 
# A function needs entry values called **arguments** (or parameters). It then performs hidden operations using these arguments and gives a **return value**.
# To use (call) a function, the command must be structured properly, following the "grammar rules" of the `R` language: the syntax.
# function_name(argument 1, argument 2)

## Arguments

# Arguments are **values** and **instructions** the function needs to run.
# Objects storing these values and instructions can be used in functions:

a <- 3
b <- 5
sum(a,b)

## Challenge 11
# - Create a vector `a` that contains all the numbers from 1 to 5
# - Create an object `b` with a value of 2
# - Add `a` and `b` together using the basic `+` operator and save the result in an object called `result_add`
# - Add `a` and `b` together using the `sum` function and save the result in an object called `result_sum`
# - Compare `result_add` and `result_sum`. Are they different?
# - Add 5 to `result_sum` function using the `sum` function

a <- c(1:5)
b <- 2

result_add <- a + b
result_sum <- sum(a,b)

result_add
result_sum
sum(result_sum, 5)

## Arguments

# Arguments each have a **name** that can be provided during a function call.
# If the name is not present, the order of the arguments does matter.
# If the name is present, the order does not matter.

a <- 1:100
b <- a^2
plot(a,b)
plot(b,a)
plot(x = a, y = b)
plot(y = b, x = a)

## Packages 

#To install packages on your computer, use the function `install.packages`.
# install.packages("packageName")

#Installing a package is not enough to use it. You need to load it into your workspace 
# Use the library() function 

install.packages("ggplot2")

qplot(1:10, 1:10)

library(ggplot2)

qplot(1:10, 1:10)


## Getting help 
# Searching for functions
# 
# To find a function that does something specific in your installed packages, you can use `??` followed by a search term.
# 
# Let's say we want to create a *sequence* of odd numers between 0 and 10 as we did earlier. We can search in our packages all the functions with the word "sequence" in them:

??sequence 

# OK! SO let's use the `seq` function!!
# 
# But wait... how does it work? What arguments does it need?
# 
# To find information about a function in particular, use `?`
# 

?seq

## Challenge 13 

# 1. Create a sequence of even numbers from 0 to 10 using the `seq` function.
  seq(from=0, to=10, by=2)
  seq(0,10,2)

#  2. Create a unsorted vector of your favourite numbers, then sort your vector in reverse order.

numbers <- c(2,4,22,6,26)
sort(numbers, decreasing = T)

## Challenge 14
# 
# Find the appropriate functions to perform the following operations:
#   
# - Square root
# - Calculate the mean of numbers
# - Combine two data frames by columns
# - List availables objects in your workspace

?sqrt 
?mean
?cbind
?ls 

## Some useful R websites

# - http://stats.stackexchange.com
# - https://www.zoology.ubc.ca/~schulter/R/
#   - http://statmethods.net/
#   - http://rseek.org/
#   - http://cookbook-r.com/
#   - http://cran.r-project.org/doc/contrib/Baggott-refcard-v2.pdf


## Thank you for attending!
########################### END OF SCRIPT ################################


workshop01-fr/ReferenceScriptWorkshop1 - FR.R
#### Atelier 1 du CSBQ 
#### Script R de référence 
## September 2018 

## Défi 1
## Comment lire la console
sortie <- "Ceci est la sortie"
sortie

# Utiliser R comme une calculatrice
## Additions et soustractions
1 + 1
10 - 1 

## Multiplications et divisions
2 * 2
8 / 2

## Exposants 
2^3

## Défi 2 
# Utilisez R pour effectuer l'opération suivante

2 + 16 * 24 - 56 

## Défi 3
# Utilisez R pour effectuer l'opération suivante
2 + 16 * 24 - 56 / (2 + 1) - 457

## Défi 4

# Quel est l'aire du cercle dont le rayon est de 5 cm ?
3.1416 * 5^2

# Notez que R a des constantes intégrées, telle que pi. Vous pouvez donc écrire:
pi * 5^2

## Objet 

# Les objets sont un des concepts le plus utilisé dans R.
# Vous pouvez stocker des valeurs dans un ojet nommé à l'aide de l'opérateur "<-"

nom_objet <- "valeurs"

## Nommer les objets: bonnes pratiques

# Choisir des noms courts et explicites 
# Il est recommander d'ajouter des espaces autour de l'opérateur "<-"
# Après avoir tapé le nom des objets, R retournera leur valeur 

moy_x <- (2 + 6) / 2
moy_x 

## Défi 5 
# Créez un objet avec une valeur de 1 + 1.718282 (le nombre d'Euler) et nommez-le "valeur_euler"

valeur_euler <- 1 + 1.718282
valeur_euler

## Défi 6 
# Créez un deuxième objet dont le nom commence par un chiffre. Que se passe-t-il ?

#8valeur_euler <- 8 * (1 + 1.718282)

## Types de structures de données en R 

# - Vecteurs
# - Data frames
# - Matrices, arrays et listes

## Vecteur

# - Une entitée constituée d'une liste de valeurs de même type
# - Une valeur seule est appelée une *atomic value*
# - Toutes les valeurs dans un vecteur doivent avoir le même mode (ou classe)
# * Numeric: seulement des nombres
# * Logical: entrées True/False
# * Character: Texte ou un mélange de texte et d'autres modes

# La création d'un vecteur nécessite la fonction c() // c() signifie combiner ou concaténer 

vecteur <- c("valeur1", "valeur2")

## Vecteurs numériques
num_vecteur <- c(1, 4, 3, 98, 32, -76, -4)
num_vecteur 

## Vecteurs de caractères
char_vecteur <- c("bleu", "rouge", "vert")
char_vecteur

## Vecteurs logiques ou booléens
bool_vecteur <- c(TRUE, TRUE, FALSE)
bool_vecteur

bool_vecteur2 <- c(T,T,F)
bool_vecteur2

## Défi 7 
# Créez un vecteur contenant les 5 premiers nombres impairs, à partir de 1 et nommez-le "impair"
impair <- c(1, 3, 5, 7)

## La structure d'un objet en utilisant la fonction dput()
dput(impair)

# Création d'un nouvel objet, identique au précédent
structure(c(1, 3, 5, 7))

## Nous pouvons utiliser des vecteurs pour faire des calculs
x <- c(1:5)
y <- 6 

x + y
x * y 

## Data frame 

# - Utilisé pour stocker des tables de données
# - une liste de vecteurs de même longueur
# - une colonne = une variable
# - une ligne = une observation, un site, un replicat, ...
# - Différentes colonnes peuvent avoir différents modes

## Une façon de faire pour créer un data frame
# Créez des vecteurs 

site_id <- c("A1.01", "A1.02", "B1.01", "B1.02")
pH_sol <- c(5.6, 7.3, 4.1, 6.0)
num_sp <- c(17, 23, 15, 7)
traitement <- c("Fert", "Fert", "Non_fert", "Non_fert")

# Combinez les vecteurs en utilisant la fonction data.frame()
mon_df <- data.frame(site_id, pH_sol, num_sp, traitement)
mon_df

# Structure du data frame avec la fonction dput()
dput(mon_df)

# Création d'un nouveau data frame identique au précédent à l'aide de la fonction structure()
structure(list(site_id = structure(1:4, .Label = c("A1.01", "A1.02","B1.01", "B1.02"), class = "factor"),
               pH_sol = c(5.6, 7.3, 4.1,6),
               num_sp = c(17, 23, 15, 7),
               traitement = structure(c(1L, 1L, 2L, 2L), .Label = c("Fert", "Non_fert"), class = "factor")),
          class = "data.frame", row.names = c(NA, -4L))
## Matrice, Array et Listes 

## Indexer un vecteur
# Vous pouvez utiliser l'indexation pour choisir une postion particulière au sein d'un vecteur,
# Par exemple, nous voulons obtenir la deuxième valeur de notre vecteur "impair"

impair[2]
# Nous pouvons sélectionner plusieurs positions 
impair[c(2, 4)]
# Nous pouvons utiliser l'indexation pour supprimer certaines valeurs à des positions particulières
impair[-c(1, 2)]

# Si nous selectionnons une position qui n'existe pas dans le vecteur
impair[c(1, 5)]

# Des conditions peuvent aussi être utilisées pour sélectionner des valeurs 
char_vecteur[char_vecteur == "bleu"]

## Défi 8 

# En utilisant le vecteur "num_vecteur"
# - Extrayez la 4ème valeur
# - Extrayez les 1ère et 3ème valeurs
# - Extrayez toutes les valeurs sauf les 2ème et 4ème

num_vecteur[4]
num_vecteur[c(1, 3)]
num_vecteur[c(-2, -4)]

## Défi 9 
# Explorez les différences entre ces deux lignes de code
char_vecteur == "bleu"
char_vecteur[char_vecteur == "bleu"]

## Indexer un data frame
## Défi 10 

# 1. Extrayez la colonne "num_sp" de "mon_df" et multipliez ses valeurs par les premières quatre valeurs de "num_vecteur"

mon_df$num_sp * num.vecteur[c(1:4)]
# or
mon_df[, 3] * num.vecteur[c(1:4)]

# 2. Ensuite, écrivez une déclaration logique qui vérifie si les valeurs obtenurs sont supérieures à 25
(mon_df$num.sp * num.vecteur[c(1:4)]) > 25

## Fonction 

# Une fonction est un outil pour vous simplifier la vie.
# 
# Il vous permet d'exécuter rapidement des opérations sur des objets sans avoir à écrire toutes les étapes mathématiques.
# 
# Une fonction nécessite des valeurs d'entrée appelées arguments (ou paramètres). Elle effectue ensuite des opérations cachées en utilisant ces arguments et donne une valeur de retour.
# Pour utiliser (appeler) une fonction, la commande doit être structurée correctement, en suivant les règles de grammaire du langage R, i.e. la syntaxe.
# nom_fonction(argument 1, argument 2)

## Arguments

# Les arguments sont des valeurs et des instructions nécessaires à la fonction pour être exécutée.
# Les objets stockant ces valeurs et ces instructions peuvent être utilisés dans des fonctions.

a <- 3
b <- 5
sum(a, b)

## Défi 11


# Créez un vecteur "a" qui contient tous les nombres de 1 à 5
# Créez un objet "b" qui a une valeur de 2
# Additionnez "a" et "b" en utilisant l'opérateur de base + et enregistrez le résultat dans un objet appelé "resultat_add"
# Additionnez "a" et "b" ensemble en utilisant la fonction sum() et enregistrez le résultat dans un objet appelé "resultat_sum"
# Comparez "resultat_add" et "resultat_sum". Sont-ils différents?
# Additionnez 5 à "resultat_sum" en utilisant la fonction sum()

a <- c(1:5)
b <- 2

resultat_add <- a + b
resultat_sum <- sum(a, b)

resultat_add
resultat_sum
sum(resultat_sum, 5)

## Arguments

# Les arguments ont chacun un **nom** pouvant être spécifié lors d'un appel de fonction.
# Si le nom n'est pas spécifié, l'ordre des arguments est important.
# Si le nom est spécifié, l'ordre n'a pas d'importance.

a <- 1:100
b <- a^2
plot(a, b)
plot(b, a)
plot(x = a, y = b)
plot(y = b, x = a)

## Package

# Pour installer des packages sur votre ordinateur, utilisez la fonction install.packages().
# install.packages("nom_du_package")

# L'installation d'un package ne suffit pas pour l'utiliser. Vous devez aussi le charger dans votre espace de travail à l'aide de la fonction library().

install.packages("ggplot2")

qplot(1:10, 1:10)

library(ggplot2)

qplot(1:10, 1:10)


## Obtenir de l'aide
# Recherche de fonctions
# 
# Pour trouver une fonction qui fait une action spécifique dans vos packages installés, vous pouvez utiliser "??" suivi d'un terme de recherche.
# 
# Disons que nous voulons créer une séquence de nombres impairs compris entre 0 et 10, comme nous l'avons fait précédemment. Nous pouvons rechercher dans nos packages toutes les fonctions qui contiennent le mot séquence dans leur description:

??sequence 

# OK! Utilisons le fonction seq() !!
# 
# Mais attendez... Comment fonctionne_t_elle ? De quels arguments a_t_elle besoin ?
# 
# Pour trouver des informations sur une fonction en particulier, utilisez "?"
# 

?seq

## Défi 13 

# 1. Créez une séquence de nombres pairs entre 0 et 10 en utilisant la fonction seq()
seq(from = 0, to = 10, by = 2)
seq(0, 10, 2)

#  2. Créez un vecteur non ordonné de vos nombres favoris, puis triez votre vecteur dans l'ordre inverse

nombres <- c(2, 4, 22, 6, 26)
sort(nombres, decreasing = T)

## Défi 14
# 
# Recherchez les fonctions appropriées pour effectuer les opérations suivantes:
#   
# - Racine carrée
# - Moyenne de plusieurs nombres
# - Combiner deux data frames par colonnes
# - Lister des objets disponibles dans votre espace de travail


?sqrt 
?mean
?cbind
?ls 

## Quelques sites internet utiles

# - http://stats.stackexchange.com
# - https://www.zoology.ubc.ca/~schulter/R/
# - http://statmethods.net/
# - http://rseek.org/
# - http://cookbook-r.com/
# - http://cran.r-project.org/doc/contrib/Baggott-refcard-v2.pdf


## Merci d'avoir participé !
########################### FIN DU SCRIPT ################################
